types/service.js

```diff
@@ -684,16 +708,19 @@ class Service extends Scribe {
    * @param  {Object}  actor Instance of the {@link Actor}.
    * @return {Promise}       Resolves upon successful registration.
    */
-  async _registerActor (actor) {
-    if (!actor.id) return this.error('Client must have an id.');
-
-    this.emit('message', `Registering Actor: ${actor.id} ${JSON.stringify(actor).slice(0, 32)}…`);
+  async _registerActor (actor = {}) {
+    if (!actor.id) {
+      const entity = new Actor(actor);
+      actor = merge({
```

Comment:
could use ... operator here

Reply:
Most of these should recurse down the property tree, you'll see this pattern elsewhere